bindings/matrix-sdk-ffi/src/room_list.rs

```diff
@@ -580,31 +584,47 @@ impl RoomListItem {
     ///
     /// ⚠️ Holding on to this room instance after it has been joined is not
     /// safe. Use `full_room` instead.
-    #[deprecated(note = "Please use `room_without_timeline` instead.")]
+    #[deprecated(note = "Please use `preview_room` instead.")]
     fn invited_room(&self) -> Result<Arc<Room>, RoomListError> {
-        self.room_without_timeline()
+        if !matches!(self.membership(), Membership::Invited) {
+            return Err(RoomListError::IncorrectRoomMembership {
+                expected: vec![Membership::Invited],
+                actual: self.membership(),
+            });
+        }
+        Ok(Arc::new(Room::new(self.inner.inner_room().clone())))
     }
 
-    /// Builds a `Room` FFI from a room without initializing its internal
-    /// timeline.
+    /// Builds a `RoomPreview` from a room list item. This is intended for rooms
+    /// with [`Membership::Invite`] or [`Membership::Knocked`],
```

Comment:
nit: this sentence ends with a comma, and it doesn't really translate to proper English IMO: maybe `This is intended for invited or knocked rooms.`?